### PR TITLE
/MAT/LAW158 : new anisotropic material law for fabric

### PR DESCRIPTION
**Pull request by MaciekWronski**
#### Description of the feature or the bug

development of a new anisotropic material law for fabric. Based on law58, but with modified algorithm and simplified input.  Stress-strain is defined by tabulated functions, no hysteresis in unloading.

#### Description of the changes

new code for new material law

### CODE REVIEW

File: starter/source/materials/mat/mat158/law158_upd.F [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+C   C o m m o n   B l o c k s
+C-----------------------------------------------
+#include      "param_c.inc"
+#include      "com04_c.inc"
+#include      "scr17_c.inc"
+#include      "tabsiz_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      CHARACTER*nchartitle  :: TITR
```
[the remark this answers is not in the document]
Comment (MaciekWronski) [reply starter/source/materials/mat/mat158/law158_upd.F]: fixed

File: engine/source/materials/mat/mat158/sigeps158c.F [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+      my_real
+     .    SIGNXX(NEL),SIGNYY(NEL),
+     .    SIGNXY(NEL),SIGNYZ(NEL),SIGNZX(NEL),
+     .    SIGVXX(NEL),SIGVYY(NEL),SIGVXY(NEL),
+     .    SOUNDSP(NEL),VISCMAX(NEL),ETSE(NEL)
+C-----------------------------------------------
+C   I N P U T   O U T P U T   A r g u m e n t s 
+C-----------------------------------------------
+      my_real UVAR(NEL,NUVAR), OFF(NEL)
+      TYPE(TTABLE) TABLE(*)
```
[the remark this answers is not in the document]
Comment (MaciekWronski) [reply engine/source/materials/mat/mat158/sigeps158c.F]: fixed

File: engine/source/materials/mat/mat158/sigeps158c.F [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+C   I N P U T   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER :: NEL, NUPARAM, NUVAR
+      my_real :: TIMESTEP
+      my_real  ,DIMENSION(NUPARAM) :: UPARAM
+      my_real  ,DIMENSION(NEL) :: AREA,RHO0,THKLY,OFFGG,TAN_PHI,
+     .   DEPSXX,DEPSYY,DEPSXY,DEPSYZ,DEPSZX,
+     .   EPSXX ,EPSYY,EPSXY ,EPSYZ ,EPSZX ,
+     .   SIGOXX,SIGOYY,SIGOXY,SIGOYZ,SIGOZX,SHF,ALDT
+C-----------------------------------------------
```
[the remark this answers is not in the document]
Comment (MaciekWronski) [reply engine/source/materials/mat/mat158/sigeps158c.F:107]: fixed intents

File: engine/source/materials/mat/mat158/sigeps158c.F [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+          EPSF=(HC0 * EPSFC + HT0 * EPSFT) / H0    ! eng strain coupling flex spring
+          HC  = HC0 * (EPSFC + ONE)                ! length of flex spring dir 1
+          HT  = HT0 * (EPSFT + ONE)                ! length of flex spring dir 2
+          DC  = SQRT(LC(I)*LC(I) + HC*HC)          ! fiber length in dir 1
+          DT  = SQRT(LT(I)*LT(I) + HT*HT)          ! fiber length in dir 2
+          DCC = DC - DC0
+          DTT = DT - DT0
+          HDC = HC / DC
+          HDT = HT / DT
+          FC  = FINTER(IFUNC(1),DCC,NPF,TF,FPC) * YFAC(1)    ! fiber force dir 1
```
[the remark this answers is not in the document]
Comment (MaciekWronski) [reply engine/source/materials/mat/mat158/sigeps158c.F]: I reorganized loops to apply VINTERs

File: engine/source/materials/mat/mat158/sigeps158c.F [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+        EC(I) = EXP(ETC) - ONE         ! eng strain dir 1
+        ET(I) = EXP(ETT) - ONE         ! eng strain dir 2
+        LC(I) = LC0 * (ONE + EC(I))    ! element length dir 1
+        LT(I) = LT0 * (ONE + ET(I))    ! element length dir 2 
+      ENDDO
+c                  
+c---    Coupled model (fibers in tension)
+c
+#include "vectorize.inc"
+      DO J = 1,NINDX
```
[the remark this answers is not in the document]
Comment (MaciekWronski) [reply engine/source/materials/mat/mat158/sigeps158c.F]: It was a copy from law58, I removed all double loops with INDEX, the vectorization should be easier

**Review by laurent-altr — APPROVED** [review]

**Pull request closed without merging**